src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -117,6 +117,18 @@ AppHost::AppHost() noexcept :
     {
         _BecomeMonarch(nullptr, nullptr);
     }
+
+    // Create a throttled function for updating the window state, to match the
+    // one requested by the pty. A 200ms delay was chosen because it's the
+    // typical animation timeout in Windows. This does result in a delay between
+    // the PTY requesting a change to the window state and the Terminal
+    // realizing it, but should mitigate issues where the Terminal and PTY get
+    // de-sync'd.
+    _showHideWindowThrottler.emplace(std::move(std::chrono::milliseconds(200)),
+                                     std::move([this](const bool show) -> winrt::fire_and_forget {
+                                         co_await wil::resume_foreground(_logic.GetRoot().Dispatcher());
+                                         _window->ShowWindowChanged(show);
+                                     }));

```

Comment:
You can replace this with `ThrottledFuncTrailing` from WinRTUtils, which should be included here already. Use it like so:
```cpp
_showHideWindowThrottler.emplace(
    _logic.GetRoot().Dispatcher(),
    std::chrono::milliseconds(200),
    [this](const bool show) -> winrt::fire_and_forget {
        _window->ShowWindowChanged(show);
    });
```

Reply:
BTW values that you "construct" - like the `std::chrono::milliseconds(200)` are already objects that C++ considers to be "moving". If you use `std::move` on them you make it worse, because you force them to exist on the metaphorical stack, so that they can be referenced (as a `T&&`).